### ISSUE TITLE
Add support for Perl 5.20 and/or Method::Signatures subroutines signature syntax

### ISSUE THREAD

**Issue by melan** [hurricup/Perl5-IDEA#455]
This and few other packages adds alternative `sub {}` definitions. The most important feature and BTW it was added as an experimental feature in 5.19: method signatures:

Pre 5.20 and w/o Method::Signatures sub syntax is:
```perl
sub a {
  my ($foo, $bar) = @_;
  ...
}
```

Perl 5.20 syntax:
```perl
sub a ($foo, $bar) {
  ...
}
```

Method::Signatures syntax:
```perl
func a (Int $foo, Str $bar) {
   ...
}

method b (Str $world) {
   $self->hello($world); # $self - is defined here implicitly
}
```

It should recognize those keywords and show arguments in the autocomplete suggestions.

**Comment by melan** [comment]
I added something like this into a SublimeCodeIntel, will try to port here if I find where to put those changes J

**Comment by hurricup** [comment]
Basic implementation will be done in #542 
But on Method::Signatures::Simple level atm

**Comment by hurricup** [comment]
Native perl signatures been implemented in #40

**Comment by hurricup** [comment]
Fixed
